src/test/java/com/puppycrawl/tools/checkstyle/JavaParserTest.java

```diff
@@ -119,36 +113,40 @@ public void testAppendHiddenSingleLineCommentNodes() throws Exception {
 
         final Optional<DetailAST> singleLineComment = TestUtil.findTokenInAstByPredicate(root,
             ast -> ast.getType() == TokenTypes.SINGLE_LINE_COMMENT);
-        assertWithMessage("Single line comment should be present")
+
+        if (singleLineComment.isPresent()) {
+            final DetailAST comment = singleLineComment.get();
+
+            assertWithMessage("Unexpected line number")
+                .that(comment.getLineNo())
+                .isEqualTo(13);
+            assertWithMessage("Unexpected column number")
+                .that(comment.getColumnNo())
+                .isEqualTo(0);
+            assertWithMessage("Unexpected comment content")
+                .that(comment.getText())
+                .isEqualTo("//");
+
+            final DetailAST commentContent = comment.getFirstChild();
+
+            assertWithMessage("Unexpected token type")
+                .that(commentContent.getType())
+                .isEqualTo(TokenTypes.COMMENT_CONTENT);
+            assertWithMessage("Unexpected line number")
+                .that(commentContent.getLineNo())
+                .isEqualTo(13);
+            assertWithMessage("Unexpected column number")
+                .that(commentContent.getColumnNo())
+                .isEqualTo(2);
+            assertWithMessage("Unexpected comment content")
+                    .that(commentContent.getText())
+                    .startsWith(" inline comment");
+        }
+        else {
+            assertWithMessage("Single line comment should be present")
                 .that(singleLineComment.isPresent())
                 .isTrue();
-
-        final DetailAST comment = singleLineComment.get();
```

Comment:
Please just use Optional#orElseThrow

Reply:
@nrmancuso , orElseThrow requires a Supplier argument, only default constructor is compatible.
Without arg to Ctor, we get another inspection problem [NewExceptionWithoutArguments](https://circleci-tasks-prod.s3.us-east-1.amazonaws.com/storage/artifacts/2233fdd7-79cb-481c-a64c-0df373c0d898/230b0534-41a7-4f5f-9290-089f863b38e2/0/home/circleci/project/target/inspection-results/NewExceptionWithoutArguments.xml?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQVFQINEOEY7XGRXI%2F20240423%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240423T145205Z&X-Amz-Expires=60&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEC8aCXVzLWVhc3QtMSJGMEQCIDRSSLS%2Bf33%2FpB%2BC4%2B7yPzhfc2P%2BkAtuf2yz8nTklJVPAiB%2FYDnlN3KHW6zAgaNO4N6hWmiXAYSJWKP8BxZT106XCiqrAgh4EAMaDDA0NTQ2NjgwNjU1NiIMgvGC%2BQOAOvDvDvpIKogCBZmT0l7vga0%2FYZ13jCo%2FpXg5n7aPtEH1t2%2BgYI7BRP7IxX3hxMGVO4ukWJ%2FPAiZ%2FE8lLq23daeBlaS2lWCqeIuK34hjWZ4s7mhR1B9Rk3zEzWrC8tJe%2Fh7l4oaQS38bB2JWNSj7%2BtzZZDyCuRPpX%2FAKBX1OLUqq4XihQf6U%2FhuYaDBcsoUmqvKIV94u5ECszeNEWCJaTJCrkrftWGSW9%2Bk4CaXAO2toyEdqHWJlBgOnP0wo8Y1ZZXJUzqfAZBubDTOUiyPJmrg5TYRkkCDLJOo5qBFKMQeZYxSo0p5GsHJ9cfnC3i32yhRoTG7H4t7XCFtGHcZNWecLxnHCqgFz6slBF4SSg7PEKMNCRn7EGOp4BigLwyVsbGSQbgEkco2XCRJO1BUxrRwQXmXzI%2B5lE9hJlPJkYO0s%2B%2Fb80r7%2Btn%2BR9VMi30FVXDgiK9rOgoPnVJhDtSMbPMTuYBFBxx1QMHs6xrvBPLxs9eWR%2Bj8ECZUfl5Tdu0a7eKNVKYSE6HQ0qD6k2Q9TZtuCsxbPmWzYDUhlF3dZ%2BGPgodM%2BfaOqRh0r4A6WAIKU720L16fzIuZo%3D&X-Amz-SignedHeaders=host&x-id=GetObject&X-Amz-Signature=ff6f901bd79f74bde0e8907367813a4909e9caa6c0ad7e14aecb30b984455ace)

We have to set default to NullPointerException if we want to use this method.

Reply:
Please see if the change is okay.